### DISCUSSION TITLE
Get a source unit's age thanks to the REST API

### DISCUSSION THREAD

**Discussion by benjaminchazelle** [WeblateOrg/weblate#7371] [Ideas]
Hello,

I would like to make a script able to remove unused translation keys in my code, older than a certain date.

For that, I would need to get the creation date of a unit. According to the `/weblate/trans/models/unit.py` file, this corresponds to the `timestamp = models.DateTimeField(auto_now_add=True)` field in the `class Unit`. 

This information is available in the Web UI, however, in the REST API, on [the  `GET /api/units/(int: id)/` endpoint](https://docs.weblate.org/en/latest/api.html#get--api-units-(int-id)-), this information is not exposed.

From what I understand from the codebase, it would be "enough" to add the timestamp field in the `class UnitSerializer` fields in the file `/weblate/api/serializers.py`

Could you expose this information? It would be very great :)

Thank you in advance !

**Comment by nijel** [comment]
Yes, it should be added as a read-only field.

**Reply by nijel** [reply]
I've just added it in 3a2be7a34f.

**Reply by benjaminchazelle** [reply]
Thank you very much! This is great :grinning:  

Do you know in which version it will be available?

**Reply by nijel** [reply]
In 4.12.

**Reply by benjaminchazelle** [reply]
Nice, thank you again :)
